File: src/main/java/org/micoli/php/symfony/list/RouteElementDTO.java
package org.micoli.php.symfony.list;

import com.intellij.psi.PsiElement;
import java.util.List;

public record RouteElementDTO(String uri, String name, String methods, String fqcn, PsiElement element)
        implements SearchableRecord {

    @Override
    public List<String> getSearchString() {
        return List.of(uri, name, methods, fqcn);
    }
}

File: src/main/kotlin/org/micoli/php/symfony/list/RouteElementDTO.kt
package org.micoli.php.symfony.list

import com.intellij.psi.PsiElement
import kotlinx.collections.immutable.ImmutableList
import kotlinx.collections.immutable.persistentListOf

@JvmRecord
data class RouteElementDTO(
    val uri: String,
    val name: String,
    val methods: String,
    val fqcn: String,
    val element: PsiElement
) : SearchableRecord {
    override fun getSearchString(): ImmutableList<String> {
        return persistentListOf(uri, name, methods, fqcn)
    }
}
